blocks/bricks/sequence_generators.py

```diff
@@ -136,21 +137,59 @@ def _push_allocation_config(self):
 
     @application
     def cost(self, application_call, outputs, mask=None, **kwargs):
-        """Returns generation costs for output sequences.
+        """Returns the average cost of over the minibatch.
+
+        The cost is computed by averaging the sum of per token costs for
+        each sequence over the minibatch.
+
+        .. warning::
+            Note that, the computed cost can be problematic when batches
+            consist of vastly different sequence lengths.
 
         Parameters
         ----------
         outputs : :class:`~tensor.TensorVariable`
             The 3(2) dimensional tensor containing output sequences.
-            The dimension 0 must stand for time, the dimension 1 for the
-            position on the batch.
+            The axis 0 must stand for time, the axis 1 for the
+            position in the batch.
         mask : :class:`~tensor.TensorVariable`
             The binary matrix identifying fake outputs.
 
+        Returns
+        -------
+        cost : :class:`~tensor.Variable`
+            Theano variable for cost, computed by summing over timesteps
+            and then averaging over the minibatch.
+
         Notes
         -----
         The contexts are expected as keyword arguments.
 
+        Adds average cost per sequence element `AUXILIARY`
+        variable to the computational graph with name
+        `~sequencegenerator_cost_per_sequence_element`
```

Comment:
In Blocks variables have two names: `x.name` and `x.tag.name`. In fact the second is more important, the first is assembled from various bits and piece for debugging purposes, for instance instead of `sequencegenerator` there can be the name of a particular brick. So please write simply `per_sequence_element` here.